keps/sig-storage/661-statefulset-volume-resize/README.md

```diff
@@ -0,0 +1,1112 @@
+    * It should also be possible to create tooling that will allow the user to aggregate events from multiple PVCs that were created by the same StatefulSet/template.
+* It will be inconsistent with how events are reported on the StatefulSet pod.
+  * Error events that are generated by kubelet on a pod (eg `Failed to pull image`) are not propagated to the pod's StatefulSet object.
+
+### Feedback
+
+![](./diag2.png)
+
+It is useful to provide feedback to the user about the status of the resize operation. This main benefits are
+
+* For errors that we can't properly report via an event on the StatefulSet object - a delay in feedback could prompt the user to investigate the issue, instead of leaving them oblivious to it. 
+* Even for scenarios where there are no errors, it could be useful to provide feedback to the user that the resize operation has completed or is still in progress. If the user has multiple StatefulSets and/or cluster they are resizing, this will give them the option to decide if they want to wait for the resize operation to complete before proceeding with other operations.
+* Feedback is especially useful if the resize happens as part of an automated deployment mechanism (ie a CI/CD pipeline). In such cases, the user doesn't necessarily keep an eye on the deployment process so proper feedback is really important.  
+
+We propose to provide feedback by adding a new field to the StatefulSet status object. The new field will be an array that contains
+* The name of the volume claim template
+* The current number of replicas for which the size of that PVC template is fully reconciled with the capacity of the PVC object (the `.status.capacity.storage` field) 
+* The last generation (pertains to the `.metadata.generation` field on the StatefulSet object) in which the size of that PVC template was fully reconciled with the capacity of the PVC object for all replicas.  
+
+Here's an example of what the StatefulSet status object with the new field will look like
+
+```yaml
+status:
+  availableReplicas: 3
+  collisionCount: 0
+  currentReplicas: 3
+  currentRevision: ex1-54c5bd476c
+  observedGeneration: 3
+  readyReplicas: 3
+  replicas: 3
+  updateRevision: ex1-54c5bd476c
+  updatedReplicas: 3
+  volumeClaimTemplates:
+  - finishedReconciliationGeneration: 2
+    readyReplicas: 1
+    templateName: vol1
+  - finishedReconciliationGeneration: 3
+    readyReplicas: 3
+    templateName: vol2
+```
+
+Some notes about the new fields
+* The `finishedReconciliationGeneration` field is optional and won't appear if the PVC template is newly created and isn't reconciled yet.
+* The `readyReplicas` will reset to 0 when the size of the PVC template is changed. It will increase only as the new pods become healthy and ready.
```

Comment:
right, I meant that if you do change something other than the PVC template (ie pod spec), it will reset the `readyReplicas` to 0 and `readyReplicas` will start increasing only as pods become healthy and the statefulset controller confirms that the PVC capacity matches the PVC template size.

I should clarify this, and also remove `ready` from `only as the new pods become healthy and ready` (because readiness checks are unrelated here)